### PR TITLE
[VPU] StaticShapeNMS support

### PR DESCRIPTION
**Pull request by AndrewBakalinIntel**
Ticket #-33362

### CODE REVIEW

**Comment by gladilov-gleb** [comment]
@AndrewBakalinIntel I don't see a link to internal validation at the ticket

File: inference-engine/src/vpu/common/include/vpu/ngraph/operations/static_shape_non_maximum_suppression.hpp [hunk 1]
```diff
@@ -14,7 +14,7 @@ namespace ngraph { namespace vpu { namespace op {
 
 class StaticShapeNonMaxSuppression : public ngraph::op::v3::NonMaxSuppression {
 public:
-    static constexpr NodeTypeInfo type_info{"StaticShapeStaticShapeNonMaxSuppression", 0};
+    static constexpr NodeTypeInfo type_info{"StaticShapeNonMaxSuppression", 0};
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/common/include/vpu/ngraph/operations/static_shape_non_maximum_suppression.hpp:17]: Nice fix! 👍

File: inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                                  {DataType::FP16},
+                                  {DataType::S32},
+                                  {DataType::FP16},
+                                  {DataType::FP16}},
+                                 {{DataType::S32},
+                                  {DataType::S32}});
+    }
+
+    void finalCheckImpl() const override {
+    }
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp:50]: While this place definitely needs refactoring - please check the same invariants at the end as well

**Comment by AndrewBakalinIntel** [comment]
> @AndrewBakalinIntel I don't see a link to internal validation at the ticket

@gladilov-gleb Sry, forgot to mark this as draft, as some changes are also required.

File: inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        "StaticShapeNMS parsing failed, expected number of outputs: 2, but {} provided",
+        outputs.size());
+
+    const auto sortResultDescending = layer->GetParamAsBool("sort_result_descending");
+    const auto boxEncoding = layer->GetParamAsString("box_encoding");
+
+    VPU_THROW_UNLESS(sortResultDescending == false,
+        "StaticShapeNMS: parameter sortResultDescending=true is not supported on VPU");
+    VPU_THROW_UNLESS(boxEncoding == "corner" || boxEncoding == "center",
+        "StaticShapeNMS: boxEncoding currently supports only two values: \"corner\" and \"center\"");
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp:93]: Please, provide actual value

File: inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+    DataVector tempInputs = inputs;
+    for (size_t fake = inputs.size(); fake < 5; fake++) {
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp:96]: Nitpick: `fake < 5` - I expect signed/unsigned comparison warning here

File: inference-engine/tests/functional/plugin/myriad/single_layer_tests/static_shape_nms.cpp [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        const auto inputScores = std::make_shared<ngraph::opset3::Parameter>(
+                ngPrc, ngraph::Shape({static_cast<size_t>(numBatches), static_cast<size_t>(numClasses), static_cast<size_t>(numBoxes)}));
+        const auto maxOutputBoxesPerClassConst = std::make_shared<ngraph::opset3::Constant>(
+                ngraph::element::i64, ngraph::Shape{}, maxOutputBoxesPerClass);
+        const auto iouThresholdConst = std::make_shared<ngraph::opset3::Constant>(
+                ngraph::element::f32, ngraph::Shape{}, iouThreshold);
+        const auto scoreThresholdConst = std::make_shared<ngraph::opset3::Constant>(
+                ngraph::element::f32, ngraph::Shape{}, scoreThreshold);
+
+        const auto staticShapeNMS = std::make_shared<ngraph::vpu::op::StaticShapeNonMaxSuppression>(
```
Comment (gladilov-gleb) [inline inference-engine/tests/functional/plugin/myriad/single_layer_tests/static_shape_nms.cpp:83]: Is there any test on dynamic NMS transformed to SSNMS?

File: inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp (hunk 3 above)
Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp]: Done.

File: inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp (hunk 4 above)
Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp]: Actually no warning there.

File: inference-engine/tests/functional/plugin/myriad/single_layer_tests/static_shape_nms.cpp (hunk 5 above)
Comment (AndrewBakalinIntel) [reply inference-engine/tests/functional/plugin/myriad/single_layer_tests/static_shape_nms.cpp:83]: I have added them, but they are currently disabled due to absence of evaluate functions for NMS - #-30919

File: inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp (hunk 2 above)
Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/stages/static_shape_nms.cpp]: Fixed.

**Comment by AndrewBakalinIntel** [comment]
@gladilov-gleb It's still waiting for final review of device side changes (in order to update the firmware). I will let you know once it's ready.

**Comment by AndrewBakalinIntel** [comment]
@gladilov-gleb could you please merge? Precommit and internal CI are fine.

**Pull request merged**